### PR TITLE
Allow users to auth via an access token

### PR DESCRIPTION
**Pull request by jetherton**

### CODE REVIEW

File: config.template.js [hunk 1]
```diff
@@ -17,11 +17,19 @@ config.source_tables = {};
 config.account.api_key = '8675309';
 
 //The name of the user to login as
+//Use username and password if you're
+//not using an access token to authorize this
+//microservice. 
 config.account.username = 'user@user.com';
 
 //The password of the user to login as
 config.account.password = 'correcthorsebatterystaple';
```
Comment (mscherer11) [inline config.template.js]: Assuming this is a placeholder

File: config.template.js [hunk 2]
```diff
@@ -17,11 +17,19 @@ config.source_tables = {};
 config.account.api_key = '8675309';
```
Comment (mscherer11) [inline config.template.js]: Placeholder?

File: index.js [hunk 3]
```diff
@@ -63,27 +63,51 @@ exports.handler = function(event, context, callback) {
...
-        return api.getView(viewId, {"start": 0, "max": 1000})
-    }).then((response) =>{
+
+    //check really hard for a valid access token
+    if(config.account.access_token 
+        && typeof config.account.access_token === 'string' 
+        && config.account.access_token.length > 20){
+            log.log("Access token seems valid, using that to authorize");
+        //access token
+        api.setAccessToken(config.account.access_token);
```
Comment (mscherer11) [inline index.js]: Should this be wrapped in a Promise to ensure that the token is set first (prob so unlikely its not something I should be worrying about)?

**Review by m-scherer — COMMENTED** [review]
Just a few comments that probably don't need to be addressed

File: config.template.js (hunk 2 above)
Comment (jetherton) [inline config.template.js]: Yes, I could change it to be more explicit like `123456` or just ``

File: config.template.js (hunk 1 above)
Comment (jetherton) [inline config.template.js]: Yes it is. And if you're not getting the reference, we'll need to talk: https://xkcd.com/936/

File: index.js (hunk 3 above)
Comment (jetherton) [inline index.js]: So I thought about that, to make sure login and setAccessToken are treated the same. But login requires talking to the server, whereas setting the access token is done locally in memory synchronously, so there's nothing async about the action. So I left it the way it is.

**Pull request merged**